internal/cli/util/util.go

```diff
@@ -0,0 +1,13 @@
+package util
```

Comment:
Nit: "util" packages are something I try to avoid because they have a tendency to become "junk drawers" for every little thing that we can't decide where else to put it.

Note: We already have a few such packages that are "util" packages in all but name and long-term, I'd like to untangle those.

I'd rather see this in a package like `cli/io` or something, because that name will make it less of a magnet for homeless code.

Better yet...

I recognize this `cli/util` package only even exists because of the need to share certain functionality across all commands, which are _each_ in their own package...

I would propose that the packages under `internal/cli` are mostly vanity packages (_and_ micro-packages) and we would do well to collapse them into a single package, in which case, shared functionality like `SetIOStreams()` no longer requires its own package to be shared across all the commands.

I'll still say this is a "nit" because I wouldn't let it block this PR, and I think any package re-structuring could happen in a follow-up PR instead of piggy-backing on this one.

Reply:
> Nit: "util" packages are something I try to avoid because they have a tendency to become "junk drawers" for every little thing that we can't decide where else to put it.

Yes, I agree and your `io` suggestion is actually a better way of dealing with this. The primary reason I put it into a separate place was because I didn't want to continue to pollute the `option` package which historically appeared to have been used as a "there is no better place for this" dump yard.

> [..] we would do well to collapse them into a single package [..]

The fear I have with this is that all sub-packages together may be a bit too much to continue to offer a clear separation of concerns, which could potentially open us up a temptation to not factor things out at all but rather move things into private functions. Which long-term is not a great solution either.

Given this, I will rename it to `io` before merge and let the idea simmer for a bit.

Reply:
> option package which historically appeared to have been used as a "there is no better place for this" dump yard.

Yep. One example of existing packages that are "util" in all but name.

> The fear I have with this is that all sub-packages together may be a bit too much to continue to offer a clear separation of concerns, which could potentially open us up a temptation to not factor things out at all but rather move things into private functions

I agree there is some logical limit to how much one lets a package sprawl before deciding it _should_ be broken up. That limit is for sure subjective. imho, the number of commands we're dealing with doesn't approach that limit.

As for common things being factored out into private functions (or types) -- I don't think that's _inherently_ terrible. 🤷‍♂️ I think there are a lot of concerns that are exclusive to a _class_ of problem -- e.g. the CLI options are exclusively a concern for CLI commands -- and keeping things of that nature close to the one place they're used feels reasonable. It doesn't preclude using other approaches for things deemed more broadly reusable.

> and let the idea simmer for a bit

💯

There's room for more discussion perhaps, and at any rate, I see no reason to rush into that sort of decision.